tests/test_mylog.py

```diff
@@ -1,255 +1,683 @@
+import base64
+import os
 import secrets
+import sys
+from time import time as get_unix_time
+from typing import Callable, Sequence, Tuple, TypeVar, Union
 
 import pytest
+import termcolor
 
 import mylog
 
+T = TypeVar("T")
 
-def test_nolock_enter():
-    nl = mylog.NoLock()
 
-    enter_rv = nl.__enter__()
+def _randint(a: int, b: int) -> int:
+    # random.randint is used only here,
+    # so we have to use "nosec" (vvvvvvvvvvvv) only once
+    return secrets.SystemRandom().randint(a, b)  # nosec B311
 
-    assert isinstance(enter_rv, bool) or enter_rv == 1
 
+def _randbytes(length: int) -> bytes:
+    # random.randbytes is used only here,
+    # so we have to use "nosec" only once
+    return secrets.token_bytes(length)  # nosec B311
 
-def test_nolock_exit():
-    nl = mylog.NoLock()
 
-    exit_rv = nl.__exit__(None, None, None)
+def _randchoice(seq: Sequence[T]) -> T:
+    # random.choice is used only here,
+    # so we have to use "nosec" only once
+    return secrets.SystemRandom().choice(seq)  # nosec B311
 
-    assert exit_rv is None
+
+def _random_int(
+    neg_ok: bool, *, only_if: Callable[[int], bool] = lambda _: True
+) -> int:
+    # sourcery skip: assign-if-exp, reintroduce-else
+    if neg_ok:
+        rv = _randint(-50, 50)
+    rv = _randint(0, 100)
+    if only_if(rv):
+        return rv
+    return _random_int(neg_ok, only_if=only_if)  # pragma: no cover
+
+
+def _random_bytes(
+    *, only_if: Callable[[int], bool] = lambda _: True
```

Comment:
```suggestion
    *, only_if: Callable[[bytes], bool] = lambda _: True
```
The `only_if` callable is called with `bytes`, not `int`.